src/screens/Products/ProductDetail.js

```diff
@@ -0,0 +1,266 @@
+import React, { Component } from 'react'
+import {
+  Button,
+  ProfilePicture,
+  Wallpaper,
+  Card,
+  CardSection
+} from '@components'
+import { Text, ScrollView, StyleSheet, LayoutAnimation } from 'react-native'
+import { withMappedNavigationProps } from 'react-navigation-props-mapper'
+import { Icon } from 'react-native-elements'
+
+class ProductDetail extends Component {
+  static navigationOptions = {
+    title: 'Product Detail'
+  };
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      showModel: false,
+      drawerOpen: false,
+      editExpanded: true,
+      statsExpanded: true
+    }
+
+    this.renderEditButton = this.renderEditButton.bind(this)
+    this.navigateToEditProduct = this.navigateToEditProduct.bind(this)
+  }
+
+  componentDidUpdate() {
+    LayoutAnimation.easeInEaseOut()
+  }
+
+  navigateToEditProduct() {
+    console.log("Button Pressed")
+  }
+
+
+  renderEditButton() {
+    if (this.state.editExpanded) {
+      return (
+        <CardSection style={{ backgroundColor: 'white' }}>
+          <Button
+            style={{ height: 20, backgroundColor: 'white' }}
+            title="Edit"
+            textColor="orange"
+            onPress={() => this.navigateToEditProduct()}
+          />
+          <Button
+            style={{ height: 25, backgroundColor: 'white' }}
+            title="Message"
+            textColor="orange"
+            onPress={() => console.log('Pressed message button')}
+          />
+        </CardSection>
+      )
+    }
+  }
+  render() {
+    return (
+      <Wallpaper style={styles.container}>
+        <ScrollView style={{ flex: 1.8 }}>
+          <Card>
+            <CardSection
+              style={{
+                backgroundColor: 'rgb(71, 77, 84)',
+                justifyContent: 'space-between'
+              }}
+            >
+              <Text style={styles.headerText}>{this.props.Title}</Text>
+              <Icon
+                name="dots-vertical"
+                size={30}
+                color="orange"
+                type="material-community"
+                underlayColor={'rgb(71, 77, 84)'}
+                onPress={() =>
+                  this.setState({ editExpanded: !this.state.editExpanded })
+                }
+              />
+            </CardSection>
+            <CardSection>
+              <ProfilePicture
+                source={{ uri: this.props.mainPictureURL }}
+                style={styles.image}
+              />
+            </CardSection>
+            {this.renderEditButton()}
+          </Card>
+          <Card>
+            <CardSection
+              style={{
+                backgroundColor: 'rgb(71, 77, 84)',
+                justifyContent: 'space-between'
+              }}
+            >
+              <Text style={styles.headerText}>{this.props.MainCategory}</Text>
+            </CardSection>
+            <CardSection
```

Comment:
this render method seems really long to me which makes it hard to read. Maybe split each Card into its own render method like you did with the edit button